app/controllers/patients_controller.rb

```diff
@@ -44,6 +40,7 @@ def update
     @patient = @hospital.patients.find(params[:id])
 
     if @patient.update(patient_params)
+      PatientBed.create!(patient_bed_params)
```

Comment:
Minha sugestão é só criar o `PatientBed` caso exista uma mudança. Pensei rapidamente em como isso poderia ser feito, acho que da para refinar:

```suggestion
      current_patient_bed = @patient.current_patient_bed
      current_patient_bed.assign_attributes(patient_bed_params)
      
      if @patient.hospitalization_date_changed? || current_patient_bed.changed?
          current_patient_bed.rollback!
          PatientBed.create!(patient_bed_params)
      end
```

Reply:
Ou no `Patient`:

```ruby
class Patient < ApplicationRecord
    def update_patient_bed!(params)
        current_patient_bed = @patient.current_patient_bed
        current_patient_bed.assign_attributes(params)

        if hospitalization_date_changed? || current_patient_bed.changed?
            current_patient_bed.rollback!
            PatientBed.create!(params)
        end
    end
end
```